server/storage/mvcc/hash.go

```diff
@@ -111,7 +111,7 @@ type hashStorage struct {
 	lg     *zap.Logger
 }
 
-func newHashStorage(lg *zap.Logger, s *store) *hashStorage {
+func NewHashStorage(lg *zap.Logger, s *store) *hashStorage {
```

Comment:
This (now) exported function returns an unexported struct. According to golangci-lint (and a linter rule we still need to enable), this is not a good practice. Refer to golang/lint#210.

However, `HashStorage` is already defined as the interface. So, addressing this may require renaming either the interface or the struct :thinking:. This may open a discussion rather than an immediate action item.

Reply:
Updated to return the interface.